File: catkin_ws/src/my_robot_tutorials/scripts/robot_news_radio_transmitter.py
#!/usr/bin/env python3

import rospy 
from std_msgs.msg import String       # Import the String message type from the standard ROS message types


if __name__ == '__main__':
		rospy.init_node('robot_news_radio_transmitter')                    # Initialize a ROS node with the name 'robot_news_radio_transmitter'
		pub = rospy.Publisher("/robot_news_radio", String, queue_size=10)  # Create a publisher that publishes to the '/robot_news_radio' topic with a queue size of 10
		rate = rospy.Rate(2)                                               # Create a rate object to control the loop frequency (2 Hz)
		
   while not rospy.is_shutdown():
			msg = String()
			msg.data = "Hi, this is Tom from the Robot News Radio !"
			pub.publish( msg )    # Publish the message to the '/robot_news_radio' topic
			rate.sleep()
     
		rospy.loginfo("Node was Stopped")                                   # Log an informational message when the node is stoppe

File: catkin_ws/src/my_robot_tutorials/scripts/robot_news_radio_transmitter.cpp
#include <ros/ros.h>  // Include the ROS C++ library

int main(int argc, char **argv) {
    
    ros::init(argc, argv, "robot_news_radio_transmitter");  // Initialize the ROS node with the name "robot_news_radio_transmitter"
    
    ros::NodeHandle nh;  // Create a node handle to interact with ROS

    // Advertise the "/robot_news_radio" topic and create a publisher for the std_msgs::String message type
    ros::Publisher pub = nh.advertise<std_msgs::String>("/robot_news_radio", 10);

    ros::Rate rate(3);  // Set the loop rate to 3 Hz

    while (ros::ok()) { 
        std_msgs::String msg;  // Create a new String message
        msg.data = "Hi, this is William from The Robot News Radio! ";  // Set the message content
      
        pub.publish(msg);  // Publish the message to the "/robot_news_radio" topic
      
        rate.sleep();  // Sleep for the remaining time to maintain the 3 Hz rate
    }

}
